wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -1514,4 +1550,93 @@ private class WasmExpressionBuilder private (
     instrs += CALL(FuncIdx(WasmFunctionName.jsLinkingInfo))
     IRTypes.AnyType
   }
+
+  // ===============================================================================
+  // array
+  // ===============================================================================
+  private def genArrayLength(t: IRTrees.ArrayLength): IRTypes.Type = {
+    genTreeAuto(t.array)
+    instrs += ARRAY_LEN
+    IRTypes.IntType
+  }
+
+  private def genNewArray(t: IRTrees.NewArray): IRTypes.Type = {
+    if (t.lengths.isEmpty || t.lengths.sizeIs > t.typeRef.dimensions)
+      throw new AssertionError(
+        s"invalid lengths ${t.lengths} for array type ${t.typeRef.displayName}"
+      )
+
+    def genWithDimensions(typeRef: IRTypes.ArrayTypeRef, lengths: List[IRTrees.Tree]): Unit = {
+      val elemTy = extractArrayElemType(t.typeRef)
```

Comment:
This line is inside the recursive `genWithDimensions` but it refers to `t` which is captured from the enclosing `genNewArray`. That's suspicious. Was it meant to be the following instead?
```suggestion
      val elemTy = extractArrayElemType(typeRef)
```